examples/basic/basic.c

```diff
@@ -0,0 +1,117 @@
+/*
+ *
+ * Copyright (C) 2024 Intel Corporation
+ *
+ * Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ *
+ */
+
+#include "umf/pools/pool_scalable.h"
+#include "umf/providers/provider_os_memory.h"
+
+#include <stdio.h>
+#include <string.h>
+
+int main(void) {
+    // A result object for storing UMF API result status
+    umf_result_t res;
+
+    // Create a memory provider
+    umf_memory_provider_ops_t *provider_ops = umfOsMemoryProviderOps();
+    umf_os_memory_provider_params_t params = umfOsMemoryProviderParamsDefault();
+    umf_memory_provider_handle_t provider;
+
+    res = umfMemoryProviderCreate(provider_ops, &params, &provider);
+    if (res != UMF_RESULT_SUCCESS) {
+        printf("Failed to create provider!");
+        return -1;
+    }
+    printf("OS Memory Provider created at %p\n", (void *)provider);
+
+    // Allocate memory from memory provider
+    size_t alloc_size = 5000;
+    size_t alignment = 0;
+    void *ptr_provider = NULL;
+
+    /// Get recommended page size for a proper alignment value
+    res = umfMemoryProviderGetRecommendedPageSize(provider, alloc_size,
+                                                  &alignment);
+    if (res != UMF_RESULT_SUCCESS || alignment == 0) {
+        printf("Failed to get recommended page size!");
+        goto memory_provider_destroy;
+    }
+    printf("Recommended page size is %zu bytes\n", alignment);
+
+    /// Allocate memory
+    res =
+        umfMemoryProviderAlloc(provider, alloc_size, alignment, &ptr_provider);
+    if (res != UMF_RESULT_SUCCESS) {
+        printf("Failed to allocate memory from the provider!");
+        goto memory_provider_destroy;
+    }
+    printf("Allocated memory at %p\n", ptr_provider);
+
+    /// Free allocated memory
```

Comment:
You could do something simple with the memory here similar to what you done for the pool